src/main/webapp/assets/script.js

```diff
@@ -222,16 +225,92 @@ function initMap() {
 
   // Get all posts on the page and show them as markers.
   posts.forEach((post) => {
+    const width = getMapMarkerIconSize(post.numOfPeopleFoodWillFeed, 'width');
+    const height = getMapMarkerIconSize(post.numOfPeopleFoodWillFeed, 'height');
+    const icon = {
+      url: getMapMarkerIconUrl(post.eventStartTime),
+      scaledSize: new google.maps.Size(width, height),
+      origin: new google.maps.Point(0, 0),
+      anchor: new google.maps.Point(0, 0),
+    };
+
     new google.maps.Marker({
       position: {lat: post.location.lat, lng: post.location.long},
       map: map,
       title: post.organizationName,
+      opacity: getMapMarkerOpacity(post.eventStartTime, post.eventEndTime),
+      icon: icon,
     });
   });
 
 /* eslint-enable no-undef */
 }
 
+/**
+ * Calculates the opacity of a given map marker.
+ * @param {Date} eventStartTime - The start time of the event.
+ * @param {Date} eventEndTime - The end time of the event.
+ * @return {number} - The opacity to show the marker as.
+ */
+function getMapMarkerOpacity(eventStartTime, eventEndTime) {
+  const now = new Date();
+  const startTime = eventStartTime;
+  const endTime = eventEndTime;
+
+  // If the date is in the future, our marker should have full opacity.
+  if (startTime > now) {
+    return 1;
+  }
+
+  // If the event is over, our marker should not be on the map at all.
+  if (endTime < now) {
+    return 0;
+  }
+
+  // Else, we need to map the full range of 1 to 0 to the range of time
+  // between eventStartTime and eventEndTime.
+  const adjusted = 1 - ((now - startTime) / (endTime - startTime));
```

Comment:
Nit. What happens if endTime == startTime? Will you have validation in user input to ensure this will never happen?

Reply:
Ideally, the user shouldn't be able to create posts where the start and end time are identical, but I agree that I should add some fallback logic so that unexpected divide by 0 errors don't occur

Reply:
Upon thinking about this issue more, this line will only cause a divide-by-zero error if `endTime === startTime === now`, because that is the only case in which the above `if ... else if` block doesn't return early if `endTime === startTime`. To account for this, I could change `else if (endTime < now)` to `else if (endTime <= now)` - that way, the `else if` can catch `endTime === now`, and any case with `endTime === startTime` will never cause a divide-by-zero.